docs/dev/ExtendingTheIceInterface.md

```diff
@@ -0,0 +1,62 @@
+
+The `MurmurIceWrapper.cpp` file generates the `*_async` versions of the Ice callbacks that handle the async nature of the Ice callbacks and also
+contain the boilerplate for e.g. verification of the caller and things like this. Most importantly though these functions call the `impl_*` functions
+defined in `MurmurIce.cpp`. For instance a function called `updateCertificates` inside the `Server` class will call `impl_Server_updateCertificate`
+which has to be defined as a `static` function inside `MurmurIce.cpp`.
+
+The declaration of the async functions generated this way are contained inside `MurmurI.h`. You have to manually add the function's declaration into
+there. The easiest way to do this is to let the script generate the implementation as described above and then copy the function signature from there
+into the `Murmur.h` file (make the declaration `virtual` though).
+
+The function signature is always
+```cpp
+static void impl_<className>_<functionName>(const ::Murmur::AMD_<className>_<functionName>Ptr cb [, int server_id] [, <function arguments>]) {
+    // Implementation goes here
+
+    cb->ice_response([<function return value>]);
+}
+```
+where `<className>` has to be substituted with the name of the class the function is declared in (e.g. `Server` or `Meta`) and `<functionName>` is the
+name of the function as declared in the `Murmur.ice` file. The parts in brackets are optional and are only needed if the new function belongs to the
+`Server` class, does take any parameters or returns a value respectively. If you have used non-default types that are declared in `Murmur.ice` (e.g.
+`IdList`), you can reference them here as `::Murmur::<typeName>` (e.g. `::Murmur::IdList`).
+
+Error reporting works via the `cb->ice_exception` function and if everything went well, the function must end by calling `cb->ice_response`
+(potentially passing a value to that function that shall be returned to the caller of the function).
+
+In general it is a good idea to have a look at the existing implementation inside `MurmurIce.cpp` and take inspiration from those.
+
+Note that the implementations make heavy use of macros (e.g. `NEED_SERVER`, `NEED_CHANNEL`, etc.). These will initialize the corresponding variables
+(`server`, `channel`, etc.) based in the parameters fed into the function. For this to work it is essential that you are using the same parameter
+names as the existing implementations (e.g. `server_id` for the server's ID).
+
+If the function requires action on the server's side (beyond its public API), you have to declare a new public function in the `Server` class (this
+time the Mumble server though; not the Ice server class) defined in `Server.h` (the definitions belong to the group of other RPC functions in there -
+section marked by a comment). The implementation of this new function should then be written in `RPC.cpp`.
```

Comment:
Can you give an example? I had no idea I'd need to be leveraging `RPC.cpp` when I first started looking at the code. This seems like an important step to just be at the bottom of the document without any section headers or bold, attention-grabbing text.

Reply:
I did add one but it is probably not the best. In the end I don't consider this to be a vital step anyways. I added an explanation of why creating a function in `RPC.cpp` is preferred. If you know that, you'll find out easily whether you need to do that for the project at hand or not :shrug: